src/shared/containers/FileUploadContainer.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import { useHistory } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+import { Storage, NexusFile } from '@bbp/nexus-sdk';
+import { useNexusContext } from '@bbp/react-nexus';
+
+import FileUploader from '../components/FileUpload';
+import { RootState } from '../store/reducers';
+
+const FileUploadContainer: React.FunctionComponent<{
+  projectLabel: string;
+  orgLabel: string;
+}> = ({ orgLabel, projectLabel }) => {
+  const nexus = useNexusContext();
+  const history = useHistory();
+  const basePath = useSelector((state: RootState) => state.config.basePath);
+  const [storages, setStorages] = React.useState<Storage[]>([]);
+
+  const makeResourceUri = (resourceId: string) => {
+    return `${basePath}/${orgLabel}/${projectLabel}/resources/${encodeURIComponent(
+      resourceId
+    )}`;
+  };
+
+  const goToResource = (resourceId: string) => {
+    history.push(makeResourceUri(resourceId));
+  };
+
+  const onFileUpload = async (file: File, storageId?: string) => {
+    const formData = new FormData();
+    formData.append('file', file);
+    return await nexus.File.create(
+      orgLabel,
+      projectLabel,
+      {
+        '@id': file.name,
+        file: formData,
+        storage: storageId,
+      },
+      // TODO: fix! https://github.com/BlueBrain/nexus/issues/784
+      // @ts-ignore
+      {
+        extraHeaders: {
```

Comment:
can you remove this? we can fix it in the SDK. But his extra header isn't fixing anything. Also, not sure why there is a @ts-ignore as extraHeader is a valid option. The issue is created so we can get rid of the comment as well

Reply:
you got it!